### PR TITLE
Design patterns Task 3 by Dimitar Dimitrov

### PR DESCRIPTION
**Pull request by ddimitrov123**
Object Pool

### CODE REVIEW

File: src/com/clouway/designpatterns/task3/Car.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.designpatterns.task3;
+
+/**
+ * @author Dimitar Dimitrov <dimitar.dimitrov045@gmail.com>
+ */
+public class Car {
+  private String model;
```
Comment (mgenov) [inline src/com/clouway/designpatterns/task3/Car.java]: could be public final

File: src/com/clouway/designpatterns/task3/PoolPattern.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    this.containerSize = containerSize;
+    newObject(object, containerSize);
+  }
+
+  public void acquire() {
+    if (containerForAvailableObjects.size() != 0){
+      containerForAvailableObjects.remove(object);
+    }else {
+      try {
+        throw new OutOfObjectsException();
```
Comment (mgenov) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: I would like to see your suggestions about better name.

**Comment by mgenov** [comment]
LGTM ! 

You are free to merge this pull request after I see that there is a good suggestion for the name of the exception.

File: src/com/clouway/designpatterns/task3/PoolPattern.java (hunk 2 above)
Comment (ddimitrov123) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: I think "ThereIsNoMoreObjectsInPoolException" is a better name, is that right?

Comment (mgenov) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: Think for better one. Still not good enough.

On 1/30/15, ddimitrov123 <notifications@github.com> wrote:
>> +  private final T object;
>> +  private final int containerSize;
>> +  private List<T> containerForAvailableObjects = new ArrayList<T>();
>> +
>> +  public PoolPattern(T object, int containerSize) {
>> +    this.object = object;
>> +    this.containerSize = containerSize;
>> +    newObject(object, containerSize);
>> +  }
>> +
>> +  public void acquire() {
>> +    if (containerForAvailableObjects.size() != 0){
>> +      containerForAvailableObjects.remove(object);
>> +    }else {
>> +      try {
>> +        throw new OutOfObjectsException();
>
> I think "ThereIsNoMoreObjectsInPoolException" is a better name, is that
> right?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/clouway-incubator/design-patterns/pull/14/files#r23827554


-- 
Regards,
  Miroslav Genov

Comment (ddimitrov123) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: NoMoreResourceInPoolException?

Comment (mgenov) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: Too long name I think. Try a better one.

On 1/30/15, ddimitrov123 <notifications@github.com> wrote:
>> +  private final T object;
>> +  private final int containerSize;
>> +  private List<T> containerForAvailableObjects = new ArrayList<T>();
>> +
>> +  public PoolPattern(T object, int containerSize) {
>> +    this.object = object;
>> +    this.containerSize = containerSize;
>> +    newObject(object, containerSize);
>> +  }
>> +
>> +  public void acquire() {
>> +    if (containerForAvailableObjects.size() != 0){
>> +      containerForAvailableObjects.remove(object);
>> +    }else {
>> +      try {
>> +        throw new OutOfObjectsException();
>
> NoMoreResourceInPoolException?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/clouway-incubator/design-patterns/pull/14/files#r23827761


-- 
Regards,
  Miroslav Genov

Comment (ddimitrov123) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: OutOfResoucesException?

Comment (mgenov) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: Sounds better, but I would like to hear another better solution.

On 1/30/15, ddimitrov123 <notifications@github.com> wrote:
>> +  private final T object;
>> +  private final int containerSize;
>> +  private List<T> containerForAvailableObjects = new ArrayList<T>();
>> +
>> +  public PoolPattern(T object, int containerSize) {
>> +    this.object = object;
>> +    this.containerSize = containerSize;
>> +    newObject(object, containerSize);
>> +  }
>> +
>> +  public void acquire() {
>> +    if (containerForAvailableObjects.size() != 0){
>> +      containerForAvailableObjects.remove(object);
>> +    }else {
>> +      try {
>> +        throw new OutOfObjectsException();
>
> OutOfResoucesException?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/clouway-incubator/design-patterns/pull/14/files#r23827900


-- 
Regards,
  Miroslav Genov

Comment (ddimitrov123) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: EmptyPoolException?

Comment (mgenov) [inline src/com/clouway/designpatterns/task3/PoolPattern.java]: PoolIsEmptyException -> pool is empty exception
IndexOutOfBounsException -> index is out of bounds exception

On 1/30/15, ddimitrov123 <notifications@github.com> wrote:
>> +  private final T object;
>> +  private final int containerSize;
>> +  private List<T> containerForAvailableObjects = new ArrayList<T>();
>> +
>> +  public PoolPattern(T object, int containerSize) {
>> +    this.object = object;
>> +    this.containerSize = containerSize;
>> +    newObject(object, containerSize);
>> +  }
>> +
>> +  public void acquire() {
>> +    if (containerForAvailableObjects.size() != 0){
>> +      containerForAvailableObjects.remove(object);
>> +    }else {
>> +      try {
>> +        throw new OutOfObjectsException();
>
> EmptyPoolException?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/clouway-incubator/design-patterns/pull/14/files#r23828080


-- 
Regards,
  Miroslav Genov

**Pull request closed without merging**